packages/dds/tree/src/events/listeners.ts

```diff
@@ -0,0 +1,68 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * `true` iff the given type is an acceptable shape for a {@link Listeners | event} listener
+ * @public
+ */
+export type IsListener<TListener> = TListener extends (...args: any[]) => void ? true : false;
+
+/**
+ * Used to specify the kinds of events emitted by a {@link Listenable}.
+ *
+ * @remarks
+ * Any object type is a valid {@link Listeners}, but only the {@link IsListener | event-like} properties of that
+ * type will be included.
+ *
+ * @example
+ * ```typescript
+ * interface MyEvents {
+ *   load: (user: string, data: IUserData) => void;
+ *   error: (errorCode: number) => void;
+ * }
+ * ```
+ *
+ * @public
+ */
+export type Listeners<T extends object> = {
+	[P in (string | symbol) & keyof T as IsListener<T[P]> extends true ? P : never]: T[P];
+};
+
+/**
+ * An object which allows the registration of listeners so that subscribers can be notified when an event happens.
+ * @param TListeners - All the {@link Listeners | events} that this subscribable supports
+ *
+ * @privateRemarks
+ * `EventEmitter` can be used as a base class to implement this via extension.
+ * ```ts
+ * type MyEventEmitter = IEventEmitter<{
+ *   load: (user: string, data: IUserData) => void;
+ *   error: (errorCode: number) => void;
+ * }>
+ * ```
+ * {@link createEmitter} can help implement this interface via delegation.
+ *
+ * @sealed @public
+ */
+export interface Listenable<TListeners extends object> {
+	/**
+	 * Register an event listener.
+	 * @param eventName - the name of the event
```

Comment:
Nit (for here and other places where we aren't using `Complete sentence casing.`
```suggestion
	 * @param eventName - The name of the event.
```